include/linux/compiler.h

```diff
@@ -248,11 +248,36 @@ static __always_inline void __write_once_size(volatile void *p, void *res, int s
  * required ordering.
  */
 
-#define READ_ONCE(x) \
-	({ union { typeof(x) __val; char __c[1]; } __u; __read_once_size(&(x), __u.__c, sizeof(x)); __u.__val; })
+#ifdef CONFIG_KTSAN
+void ktsan_sync_acquire(void *addr);
+void ktsan_sync_release(void *addr);
+#else /* CONFIG_KTSAN */
+static inline void ktsan_sync_acquire(void *addr);
+static inline void ktsan_sync_release(void *addr);
+#endif
+
+#define READ_ONCE(x)					\
+({							\
+	typeof(x)* ___x1 = &(x);			\
+	union { typeof(x) __val; char __c[1]; } __u;	\
+	__read_once_size(___x1, __u.__c, sizeof(x));	\
+	/* FIXME(xairy): remove acquire, READ_ONCE	\
```

Comment:
It is more important to document the _reason_ it is here now.
I know that READ_ONCE does not imply a barrier without this comment.
You say "remove acquire", which raises a great question as to why you added this comment rather than actually removed the acquire.